### PR TITLE
feat: validate message timestamp in signed topic validator

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: waku/v2/protocol/relay/validators.go [hunk 1]
```diff
@@ -13,19 +15,46 @@ import (
...
 	"go.uber.org/zap"
 	proto "google.golang.org/protobuf/proto"
 )
 
 // Application level message hash
 func MsgHash(pubSubTopic string, msg *pb.WakuMessage) []byte {
-	// TODO: Other fields?
-	return hash.SHA256([]byte(pubSubTopic), msg.Payload, []byte(msg.ContentTopic))
+	timestampBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(timestampBytes, uint64(msg.Timestamp))
```
Comment (alrevuelta) [inline waku/v2/protocol/relay/validators.go:26]: the test vector is working fine, but `Timestamp` is a `int64` not `uint64`.

**Review by alrevuelta — APPROVED** [review]
lgtm! minor comment

File: waku/v2/protocol/relay/validators.go (hunk 1 above)
Comment (alrevuelta) [reply waku/v2/protocol/relay/validators.go:26]: well i guess it doesnt matter as long as the number is possitive (which would be). guess that a positive uint64 is represented in bytes just like an int64.

Comment (richard-ramos) [reply waku/v2/protocol/relay/validators.go:26]: `PutUint64` expects an `uint64` hence why I cast the timestamp to that type

**Pull request merged**